### PR TITLE
Support ceilometer-agent role deployemt on hyperv nodes

### PR DESCRIPTION
**Pull request by nkrinner**
Create role ceilometer-agent-hypev, adjust barclamp and
add a migration

**Note:** Depends on https://github.com/crowbar/barclamp-hyperv-compute/pull/32

### CODE REVIEW

File: chef/roles/ceilometer-agent-hyperv.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name "ceilometer-agent-hyperv"
+description "Ceilometer Agent Role on HyperV Hosts"
+run_list(
+         "recipe[hyperv::do_setup]",
+         "recipe[hyperv::do_ceilometer]"
+)
```
Comment (jsuchome) [inline chef/roles/ceilometer-agent-hyperv.rb]: Is the usage of recipes from other package reflected in some requirements?

**Comment by jsuchome** [comment]
I've applied the patch and it doesn't cause any problems, i.e. it could be merged independently on the hyperv work, as the logic prevents user to assign wrong node to the new role.

**Comment by nkrinner** [comment]
Now also checks for the presence of the hyperv cookbook if the user tries to deploy on hypev hosts.

**Comment by vuntz** [comment]
Needs a rebase.

File: chef/cookbooks/ceilometer/recipes/common.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@
   bind_host = admin_address
   bind_port = node[:ceilometer][:ha][:ports][:api]
 else
-  bind_host = "0.0.0.0"
+  bind_host = node[:ceilometer][:api][:host]
```
Comment (vuntz) [inline chef/cookbooks/ceilometer/recipes/common.rb]: Any reason for that change? (I'm not against it, just sounds irrelevant, maybe split it in a different commit but keep it in this PR)

**Comment by tboerger** [comment]
LGTM, needs rebase

File: crowbar_framework/app/models/ceilometer_service.rb [hunk 3]
```diff
@@ -127,6 +137,10 @@ def validate_proposal_after_save proposal
 
     validate_minimum_three_nodes_in_cluster(proposal)
 
+    unless proposal["deployment"]["ceilometer"]["elements"]["ceilometer-hyperv-agent".empty?] || hyperv_available?
```
Comment (jsuchome) [inline crowbar_framework/app/models/ceilometer_service.rb]: I guess this should have been ``["ceilometer-hyperv-agent"].empty?``

File: crowbar_framework/app/models/ceilometer_service.rb [hunk 4]
```diff
@@ -127,6 +137,10 @@ def validate_proposal_after_save proposal
...
+      validation_error("Hyper-V support is not available.")
```
Comment (jsuchome) [inline crowbar_framework/app/models/ceilometer_service.rb]: Could we elaborate a bit more? This means that the hyperv cookbook is not installed, right? So maybe put a short hint into that sentence

Comment (vuntz) [inline crowbar_framework/app/models/ceilometer_service.rb]: To be fair, we have the same string in the nova barclamp :-)

Comment (nkrinner) [inline crowbar_framework/app/models/ceilometer_service.rb]: We could be more verbose here, but I guess the error message gives enough information and I'd like to be consistent with the nova-compute-hyperv check.

File: chef/cookbooks/ceilometer/recipes/common.rb (hunk 2 above)
Comment (nkrinner) [inline chef/cookbooks/ceilometer/recipes/common.rb]: I did not like the hard-coded value here. I am open to revert it, this is not important to me.

Comment (vuntz) [inline chef/cookbooks/ceilometer/recipes/common.rb]: As I said, I'm fine with the change. It should just be in a separate commit.

File: crowbar_framework/app/models/ceilometer_service.rb (hunk 3 above)
Comment (nkrinner) [inline crowbar_framework/app/models/ceilometer_service.rb]: Yes.

**Comment by vuntz** [comment]
+1

File: crowbar_framework/app/models/ceilometer_service.rb [hunk 5]
```diff
@@ -119,6 +129,10 @@ def validate_proposal_after_save proposal
 
     validate_minimum_three_nodes_in_cluster(proposal)
 
+    unless proposal["deployment"]["ceilometer"]["elements"]["ceilometer-hyperv-agent"].empty? || hyperv_available?
```
Comment (ovisan) [inline crowbar_framework/app/models/ceilometer_service.rb]: undefined method `empty?' for nil:NilClass

Throws an error in the crowbar when applying the ceilometer-hyperv-agent.
latest-screenshot

Comment (vuntz) [inline crowbar_framework/app/models/ceilometer_service.rb]: Did you do `barclamp_install.rb --rpm ceilometer` and `rccrowbar force-reload`?

Comment (ovisan) [inline crowbar_framework/app/models/ceilometer_service.rb]: Yes, happens also if i do that.

Comment (nkrinner) [inline crowbar_framework/app/models/ceilometer_service.rb]: I think I got it. It has to be `ceilometer-agent-hyperv` and not `ceilometer-hyperv-agent`

**Comment by nkrinner** [comment]
Fixed a typo that caused a undefined method on nil:Nilclass error

**Comment by ovisan** [comment]
:+1:  works

**Comment by aplanas** [comment]
+1

**Comment by jsuchome** [comment]
Ouch, schema revision should be bumped, and probably use different number than 14
